log4j-core/src/main/java/org/apache/logging/log4j/core/util/BasicAuthorizationProvider.java

```diff
@@ -35,6 +38,11 @@ public class BasicAuthorizationProvider implements AuthorizationProvider {
     public static final String CONFIG_USER_NAME = "log4j2.configurationUserName";
     public static final String CONFIG_PASSWORD = "log4j2.configurationPassword";
     public static final String PASSWORD_DECRYPTOR = "log4j2.passwordDecryptor";
+    /*
+     * Properties used to specify the encoding in HTTP Basic Authentication
+     */
+    private static final String BASIC_AUTH_ENCODING = "log4j2.configurationAuthorizationEncoding";
+    private static final String SPRING_BASIC_AUTH_ENCODING = "logging.auth.encoding";
```

Comment:
I am not happy with
1. having `configuration` in a property name (isn't everything has something to do with some _configuration_?)
2. adding Spring-specific functionality to a Spring-agnostic product

But... The rest of the class has already embraced these wrong conventions. Hence, you successfully have mirrored it.

Reply:
I think these property are transitional, because:

- most password are ASCII anyway, since you need sometimes to type them on a keyboard of an OS without advanced input methods,
- most modern OSes use `UTF-8` as system encoding anyway (Windows is the exception),
- using system encodings is obsolete (besides the system encoding of the client and server might differ).

I will **drop** these configuration properties in `3.x` and leave `UTF-8` as only encoding.